### PR TITLE
 Cleanup only if something was actually deployed

### LINKED ISSUE #812: skaffold dev must not register cleanup handler until build is complete
skaffold v0.9.0

I just accidentally ran `skaffold dev` in my PROD cluster and hit Ctrl+C immediately when the first image build started. Even though nothing actually was deployed to my PROD cluster, hitting Ctrl+C triggered the "cleanup handler", which nuked deployments on my PROD cluster. 

The cleanup handler probably should not be registered until the first build is complete?

### PR DESCRIPTION
**Pull request by dgageot**
Fix #812

### CODE REVIEW

File: pkg/skaffold/runner/runner.go [hunk 1]
```diff
@@ -260,6 +266,13 @@ func (r *SkaffoldRunner) BuildAndTest(ctx context.Context, out io.Writer, artifa
 	return bRes, err
 }
 
+// Deploy deployes the given artifacts
```
Comment (nkubala) [inline pkg/skaffold/runner/runner.go]: nit: s/deployes/deploys

File: pkg/skaffold/runner/runner.go [hunk 2]
```diff
@@ -260,6 +266,13 @@ func (r *SkaffoldRunner) BuildAndTest(ctx context.Context, out io.Writer, artifa
...
+func (r *SkaffoldRunner) Deploy(ctx context.Context, out io.Writer, artifacts []build.Artifact) ([]deploy.Artifact, error) {
+	dRes, err := r.Deployer.Deploy(ctx, out, artifacts)
+	r.hasDeployed = true
```
Comment (nkubala) [inline pkg/skaffold/runner/runner.go]: I guess it's not different than what we had before, but if `Deploy()` errors out do we want to set this flag?

Comment (dgageot) [reply pkg/skaffold/runner/runner.go]: I guess it could fail and have deployed a few things. We want to clean this up

Comment (nkubala) [reply pkg/skaffold/runner/runner.go]: yeah, not really a good way to know whether or not that's the case so let's just be safe.

**Pull request merged**